src/main/java/frc/robot/Robot.java

```diff
@@ -95,109 +118,325 @@ public void autonomousInit() {
+      else if (counter == 4) {
+        // actual field distance is 230
+        System.out.println("Entered 4");
+        if (pidDrive.calculate(leftEncoder.getDistance(), 50) > 0.45) {
+          robotDrive.arcadeDrive(pidDrive.calculate(leftEncoder.getDistance(), 50), 0.0);
+        } else if (leftEncoder.getDistance() < 50) {
+          robotDrive.arcadeDrive(0.45, 0.0);
+        }
+
+        if (timer.get() > 3) {
+          System.out.println(leftEncoder.getDistance());
+          intake.set(0.0);
+          robotDrive.arcadeDrive(0.0, 0.0);
+          leftEncoder.reset();
+          gyro.reset();
+          timer.reset();
+          counter++;
+        }
       }
-    }
 
-    else if (counter == 6) {
-      robotDrive.arcadeDrive(0.0, -pidTurn.calculate(gyro.getAngle(), 180));
+      else if (counter == 5) {
+        
+        
+        if (timer.get() < 1) {
+          intake.set(0.4);
+        } else {
+          intake.set(0.0);
+          timer.reset();
+          counter++;
+          counter2 = 0;
+        }
+      }
 
-      if (timer.get() > 2) {
-        gyro.reset();
-        leftEncoder.reset();
-        robotDrive.arcadeDrive(0.0, 0.0);
-        timer.reset();
-        counter++;
+      else if (counter == 6) {
+        // robotDrive.arcadeDrive(0.0, -pidTurn.calculate(gyro.getAngle(), 180));
+
+        if (counter2 == 0) {
+
+          if (gyro.getAngle() < 177) {
+            robotDrive.arcadeDrive(0.0, -0.80);
+          } else {
+            counter2++;
+          }
+
+        }
+        if (counter2 == 1) {
+          if (gyro.getAngle() > 180.5) {
+            robotDrive.arcadeDrive(0.0, 0.52);
+            timer.reset();
+          }
+          else if (gyro.getAngle() < 179.5) {
+            robotDrive.arcadeDrive(0.0, -0.52);
+            timer.reset();
+          } 
+          else if(timer.get() < 0.5 && gyro.getAngle() > 179.5 && gyro.getAngle() < 180.5){
+            robotDrive.arcadeDrive(0.0,0.0);
+  
+          }
+
+
+          if (timer.get() > 0.5 && gyro.getAngle() > 179.5 && gyro.getAngle() < 180.5) {
+            gyro.reset();
+            leftEncoder.reset();
+            robotDrive.arcadeDrive(0.0, 0.0);
+            timer.reset();
+            System.out.println("done");
+            counter++;
+            counter2 = 0;
+          }
+
+          }
+        }
+         else if (counter == 7) {
+        // actual field distance is 230
+        if (pidDrive.calculate(leftEncoder.getDistance(), 50) > 0.45) {
+          robotDrive.arcadeDrive(pidDrive.calculate(leftEncoder.getDistance(), 50), 0.0);
+        } else if (leftEncoder.getDistance() < 50) {
+          robotDrive.arcadeDrive(0.45, 0.0);
+        }
+        //intake.set(-0.4);
+
+        if (timer.get() > 2) {
+          System.out.println(leftEncoder.getDistance());
+          intake.set(0.0);
+          robotDrive.arcadeDrive(0.0, 0.0);
+          leftEncoder.reset();
+          gyro.reset();
+          timer.reset();
+          counter++;
```

Comment:
could maybe move the reset stuff at the end of every block into its own function? so you don't have to rewrite it again and again